R/source.R

```diff
@@ -282,22 +270,65 @@ invoke_cargo <- function(toolchain, specific_target, dir, profile,
     # If RTOOLS40_HOME is properly set, this will have no real effect
     withr::local_envvar(RTOOLS40_HOME = rtools_home)
   }
-  status <- system2(
+
+  message_buffer <- character(0)
+  env <- rlang::current_env()
+
+  tty_has_colors <- isTRUE(cli::num_ansi_colors() > 1L)
+
+  compilation_result <- processx::run(
     command = "cargo",
     args = c(
       sprintf("+%s", toolchain),
       "build",
       "--lib",
-      if (!is.null(specific_target)) sprintf("--target %s", specific_target) else NULL,
-      sprintf("--manifest-path %s", file.path(dir, "Cargo.toml")),
-      sprintf("--target-dir %s", file.path(dir, "target")),
-      if (profile == "release") "--release" else NULL
+      if (!is.null(specific_target)) {
+        sprintf("--target=%s", specific_target)
+      } else {
+        NULL
+      },
+      sprintf("--manifest-path=%s", file.path(dir, "Cargo.toml")),
+      sprintf("--target-dir=%s", file.path(dir, "target")),
+      if (profile == "release") "--release" else NULL,
+      "--message-format=json-diagnostic-rendered-ansi",
+      ifelse(
```

Comment:
Why `ifelse` instead of `if () ... else`? `ifelse` is normally used for vectorized data. Also, it's inconsistent with line 292.

Reply:
Fixed this and a few other arguments. I think `{glue}` works better here than `sprintf()`.